src/compas_ui/rhino/objects/networkobject.py

```diff
@@ -0,0 +1,344 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import reduce
+from operator import mul
+
+import compas_rhino
+
+from compas.geometry import Point
+from compas.geometry import Scale
+from compas.geometry import Translation
+from compas.geometry import Rotation
+
+from ._modify import network_update_attributes
+from ._modify import network_update_node_attributes
+from ._modify import network_update_edge_attributes
+from ._modify import network_move_node
+
+from ._object import BaseObject
+
+
+class NetworkObject(BaseObject):
+    """Class for representing COMPAS networkes in Rhino.
+
+    Parameters
+    ----------
+    network : :class:`compas.datastructures.Network`
+        A network data structure.
+    scene : :class:`compas.scenes.Scene`, optional
+        A scene object.
+    name : str, optional
+        The name of the object.
+    layer : str, optional
+        The layer for drawing.
+    visible : bool, optional
+        Toggle for the visibility of the object.
+    settings : dict, optional
+        A dictionary of settings.
+
+    """
+
+    SETTINGS = {
+        'color.nodes': (255, 255, 255),
+        'color.edges': (0, 0, 0),
+        'show.nodes': True,
+        'show.edges': True,
+        'show.nodelabels': False,
+        'show.edgelabels': False,
+    }
+
+    def __init__(self, network, scene=None, name=None, layer=None, visible=True, settings=None):
+        super(NetworkObject, self).__init__(network, scene, name, layer, visible)
+        self._guid_node = {}
+        self._guid_edge = {}
+        self._guid_nodelabel = {}
+        self._guid_edgelabel = {}
```

Comment:
we should implement a two-way dict for this, so the lookups can be done in both directions...